src/main/java/uk/gov/ons/ctp/response/action/scheduled/distribution/ActionDistributor.java

```diff
@@ -95,26 +92,39 @@ public DistributionInfo distribute() {
     return Arrays.asList(requestCount, cancelCount);
   }
 
+  private List<Action> retrieveActions(final ActionType actionType) {
+    return actionRepo.findSubmittedOrCancelledByActionTypeName(
+        actionType.getName(), appConfig.getActionDistribution().getRetrievalMax());
+  }
+
   private void processActions(
       final List<Action> actions,
       final InstructionCount requestCount,
       final InstructionCount cancelCount) {
-    log.debug(
-        "Dealing with actions {}",
-        actions.stream().map(Objects::toString).collect(Collectors.joining(",")));
-    for (final Action action : actions) {
+    for (Action action : actions) {
       try {
         processAction(action, requestCount, cancelCount);
-      } catch (final Exception e) {
-        log.error(
-            "Could not processing action {}."
-                + " Processing will be retried at next scheduled distribution",
-            action.getId(),
-            e);
+      } catch (Exception e) {
+        log.with("actionId", action.getId())
+            .error("Could not process action. Will be retried at next schedule");
       }
     }
   }
 
+  private void processAction(
+      Action action, final InstructionCount requestCount, final InstructionCount cancelCount)
```

Comment:
don't mix final/non-final unless you're genuinely intending on changing the value of action inside this function (note: final does not mean immutable)